xapian-letor/scorer/ndcg_score.cc

```diff
@@ -0,0 +1,76 @@
+/** @file ndcg_score.cc
+ *  @brief Implementation of NDCGScore
+ */
+/* Copyright (C) 2014 Hanxiao Sun
+ * Copyright (C) 2016 Ayush Tomar
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
```

Comment:
Every `.cc` source file (but NB **not** `.h` files) should `#include <config.h>` as the first header just after the copyright boilerplate.  This means that any `HAVE_XXX` preprocessor defines set as a result of `configure` checks will be available (both to this source file and to any headers that get included).